File: src/main/java/org/azamov/learnjakarta/task7_1/filter/AuthFilter.java
package org.azamov.learnjakarta.task7_1.filter;

import java.io.IOException;
import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.annotation.WebFilter;
import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@WebFilter(urlPatterns = {"/*"})
public class AuthFilter implements Filter {

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        HttpServletRequest req = (HttpServletRequest) request;
        HttpServletResponse resp = (HttpServletResponse) response;
        String path = req.getServletPath();

        if ("/login".equals(path) || "/register".equals(path)) {
            chain.doFilter(request, response);
            return;
        }

        Cookie[] cookies = req.getCookies();
        Cookie userCookie = null;

        if (cookies != null) {
            for (Cookie cookie : cookies) {
                if ("userId".equals(cookie.getName())) {
                    userCookie = cookie;
                    break;
                }
            }
        }

        if (userCookie == null || userCookie.getValue() == null) {
            resp.sendRedirect(req.getContextPath() + "/login");
        } else {
            chain.doFilter(request, response);
        }
    }
}

File: src/main/java/org/azamov/learnjakarta/task3/filter/AuthFilter.kt
package org.azamov.learnjakarta.task3.filter

import javax.servlet.*
import javax.servlet.annotation.WebFilter
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@WebFilter(urlPatterns = ["groups", "students"])
class AuthFilter : Filter {
    override fun doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain) {
        val req = request as HttpServletRequest
        val resp = response as HttpServletResponse
        val userCookie = req.cookies.find { it.name == "userId" }
        if (userCookie == null || userCookie.value == null) {
            resp.sendRedirect(req.contextPath + "login")
        } else {
            chain.doFilter(request, response)
        }
    }
}
